mlir/lib/Dialect/Arith/Transforms/IntRangeOptimizations.cpp

```diff
@@ -184,6 +189,232 @@ struct DeleteTrivialRem : public OpRewritePattern<RemOp> {
   DataFlowSolver &solver;
 };
 
+static Type checkArithType(Type type, unsigned targetBitwidth) {
```

Comment:
This name is not very descriptive. What is being checked and what does it return?